### PR TITLE
ManagementTipsAndTricksController "Update"

### PR DESCRIPTION
**Pull request by Yura09**

### CODE REVIEW

File: core/src/main/java/greencity/dto/tipsandtricks/TipsAndTricksDtoManagement.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @Size(min = 1, max = 170, message = "size must be between 1 and 170")
+    private String title;
+
+    @Size(min = 20, max = 63206, message = "size must be between 20 and 63206")
+    private String text;
+
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+    private ZonedDateTime creationDate;
+
+    @Email
```
Comment (antonbakalets) [inline core/src/main/java/greencity/dto/tipsandtricks/TipsAndTricksDtoManagement.java:39]: According to https://javaee.github.io/javaee-spec/javadocs/javax/validation/constraints/Email.html the default regexp pretty weak. You need to add your own regexp.

Ask @yuraolkhovskyi for more details, he already solved this problem.

File: core/src/main/resources/templates/core/update_tips_and_tricks.html [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+</head>
+
+<body>
+<div id="header" th:insert="core/header"></div>
+<div class="main-content">
+    <form  th:action="@{/management/tipsandtricks/edit/{id}(id=${tipsAndTricksDtoManagement.id})}"
+            th:object="${tipsAndTricksDtoManagement}"
+            method="post"
+          enctype="multipart/form-data">
+        <table class="table">
```
Comment (antonbakalets) [inline core/src/main/resources/templates/core/update_tips_and_tricks.html:31]: Is that a project convention to use <table> to format forms elements location on the page?

**Review by antonbakalets — CHANGES_REQUESTED** [review]
Please add email regexp.

File: core/src/main/java/greencity/dto/tipsandtricks/TipsAndTricksDtoManagement.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Size(min = 20, max = 63206, message = "size must be between 20 and 63206")
+    private String text;
+
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+    private ZonedDateTime creationDate;
+
+    @NotBlank(message = EMPTY_EMAIL)
+    @Email(
+        regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,6}$",
```
Comment (antonbakalets) [inline core/src/main/java/greencity/dto/tipsandtricks/TipsAndTricksDtoManagement.java:45]: There is already a constant defined for that.

**Review by antonbakalets — APPROVED** [review]

**Review by TarasDovganyuk — APPROVED** [review]

**Review by eyouu — APPROVED** [review]

**Pull request merged**